src/tunnelbridge_cmd.cpp

```diff
@@ -1682,6 +1716,7 @@ void DrawBridgeMiddle(const TileInfo *ti)
 	/* Do not draw anything more if bridges are invisible */
 	if (IsInvisibilitySet(TO_BRIDGES)) return;
 
+	if (blocked_pillars.Any(pillars)) return;
```

Comment:
Can a bridge middle tile have more than one pillar? (One for each side, perhaps) If I'm reading this right, the tile gets no pillars if any are blocked. I could certainly be reading wrong or misunderstanding something fundamental about pillars. 🙂

Reply:
Basically I didn't bother making this more complex because a bridge with pillars drawn only on one side also looks very odd.